lib/jormungandr/test/integration/Cardano/Wallet/Jormungandr/NetworkSpec.hs

```diff
@@ -341,10 +341,19 @@ spec = do
             let signed = mkTx [] outs
             runExceptT (postTx nw signed) `shouldReturn` Right ()
 
+        it "invalid binary should fail early" $ \(nw, _) -> do
+            let signed = SealedTx $ unsafeFromHex "0000"
+            r <- runExceptT (postTx nw signed)
+            case r of
+                Right _ -> expectationFailure "invalid tx should fail"
+                Left _ -> return ()
+
         it "encoder throws an exception if tx is invalid (eg too many inputs)" $
             \(nw, _) -> do
                 let inps = replicate 300 (head $ fst txNonEmpty)
                 let outs = replicate 3 (head $ snd txNonEmpty)
+                -- Previously it was postTx that could throw. Now it is mkTx.
+                -- But because of laziness this test still works!
```

Comment:
😕 

could probably be possible to remove the throwing with real error handling now, if desired. But don't think we have to do that now.